datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -143,6 +145,7 @@ impl PruningPredicate {
             schema,
             predicate_expr,
             required_columns,
+            logical_expr: expr.clone(),
```

Comment:
It might be worth considering changing the signature of this function from `

```rust
pub fn try_new(expr: &Expr, schema: SchemaRef) -> Result<Self> {
``` 

to take the Expr directly rather than `clone` within 

```rust
pub fn try_new(expr: Expr, schema: SchemaRef) -> Result<Self> {
```

The rationale would be if the caller already has a `Expr` they could pass it in rather than forcing a copy.

Reply:
We could always do that as a follow on PR as well

Reply:
Makes sense. I should have some time tomorrow to see if I can do this relatively easily. If it turns out to be more involved then we can create a follow up task for it.

Reply:
Thanks @thinkharderdev  -- to keep the code flowing I am going to merge this PR as is and we can do the cleanup in a follow on PR. Let me know if you don't get a chance and I can file a follow on ticket to do so